nixos/tests/hostname.nix

```diff
@@ -56,13 +56,6 @@ let
         )
         assert "localhost" == machine.succeed("getent hosts ::1 | awk '{print $2}'").strip()
 
-        # 127.0.0.2 should resolve back to the FQDN and hostname:
-        fqdn_and_host_name = "${optionalString (domain != null) "${hostName}.${domain} "}${hostName}"
-        assert (
-            fqdn_and_host_name
-            == machine.succeed("getent hosts 127.0.0.2 | awk '{print $2,$3}'").strip()
-        )
-
         assert "${fqdn}" == machine.succeed("getent hosts ${hostName} | awk '{print $2}'").strip()
```

Comment:
It's questionable whether this line should remain. The only reason why this works is that the testing framework's base configuration includes an `extraHosts` setting that includes this machine's FQDN:

https://github.com/NixOS/nixpkgs/blob/c086623a9d504d4b6160398f92e6daa91cc60426/nixos/lib/testing/network.nix#L89-L110

So this is just testing the test framework itself, and not really the network configuration for regular hosts.